nixos/modules/programs/rust-motd.nix

```diff
@@ -27,10 +55,43 @@ in {
         For possible formats, please refer to {manpage}`systemd.time(7)`.
       '';
     };
+    order = mkOption {
+      type = types.listOf types.str;
+      default = attrNames cfg.settings;
+      defaultText = literalExpression "attrNames cfg.settings";
+      description = mdDoc ''
+        The order of the sections in [](#opt-programs.rust-motd.settings) implies
+        the order of sections in the motd. Since attribute sets in Nix are always
+        ordered alphabetically internally this means that
+
+        ```nix
+        {
+          uptime = { /* ... */ };
+          banner = { /* ... */ };
+        }
+        ```
+
+        will still have `banner` displayed before `uptime`.
+
+        To work around that, this option can be used to define the order of all keys,
+        i.e.
+
+        ```nix
+        {
+          order = [
+            "uptime"
+            "banner"
+          ];
+        }
+        ```
+
+        makes sure that `uptime` is placed before `banner` in the motd.
```

Comment:
I guess it could be a bit more extensive actually, this feels a bit too short now, but I think you get the idea :)

Reply:
> I think this should be worded a bit more like a feature

The first question people who don't know about how Nix orders attribute-sets internally will ask is "why"? I'm pretty sure that in this case it's pretty counter-intuitive that `{ uptime = {}; banner = {}; }` doesn't result in

```toml
[uptime]
# ...
[banner]
# ...
```

if you don't know enough about how the Nix language works.

So while I agree that depending on the (apparently undefined) ordering of sections in TOML isn't very nice, I see this mostly as a workaround because of Nix's internals in this regard.

Reply:
Fair, I think it would be nice for the first paragraph to be a brief explanation of what the option does, I like this:

```
The order of the sections in [](#opt-programs.rust-motd.settings).
By default they are ordered alphabetically.
```

After that could be more context.

But this is just a minor thing, I'm not gonna block on this :)

Reply:
Done!